app/helpers/email_signup_helper.rb

```diff
@@ -0,0 +1,24 @@
+module EmailSignupHelper
+
+  def different_email_signup_path(atom_url)
+    special_case = special_cases.select { |c| atom_url.match(c[:regex]) }.first
```

Comment:
You can use [`Enumerable#find`](http://ruby-doc.org/core-2.1.4/Enumerable.html#method-i-find) instead of `select { … }.first` here